src/Symfony/Component/Validator/Constraints/SemVerValidator.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ */
+class SemVerValidator extends ConstraintValidator
+{
+    /**
+     * Semantic Versioning 2.0.0 regex pattern.
+     * Supports: 1.0.0, 1.2.3, 1.2.3-alpha, 1.2.3-alpha.1, 1.2.3+20130313144700, 1.2.3-beta+exp.sha.5114f85
+     * With optional "v" prefix: v1.0.0, v1.2.3-alpha
+     */
+    private const SEMVER_PATTERN = '/^'
+        .'(?P<prefix>v)?'                                                // Optional "v" prefix
```

Comment:
The spec says that `v...` is **not** a valid semver:

> Is “v1.2.3” a semantic version?
>
> No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control. Example: git tag v1.2.3 -m "Release version 1.2.3", in which case “v1.2.3” is a tag name and the semantic version is “1.2.3”.

See https://semver.org/#is-v123-a-semantic-version

-----

Should we add a config option to enable strict validation following the spec?

Reply:
I like the `strict` option idea

Reply:
implemented